pkg/gotohelm/transpiler.go

```diff
@@ -148,7 +148,11 @@ func (t *Transpiler) transpileFile(f *ast.File) *File {
 
 		funcDirectives := parseDirectives(fn.Doc.Text())
 		if _, ok := funcDirectives["name"]; !ok {
-			funcDirectives["name"] = fn.Name.String()
+			if fn.Recv != nil {
+				funcDirectives["name"] = fmt.Sprintf("%s.%s", fn.Recv.List[0].Type.(*ast.StarExpr).X.(*ast.Ident), fn.Name.String())
```

Comment:
> This cast is pretty brittle. It might be better to use typeOf here to get the function type as it'll be a bit easier to resolve the receiver to a named type.

I totally agree it's brittle. In the next PR it is a bit improved https://github.com/redpanda-data/helm-charts/pull/1343/files#diff-5fd16cbd6eecffe68e88c97a779873344d516695142c55bb3a83e55653253df8R151-R157

> We can keep it as is for now but if it ever causes us grief, we should go down the type route rather than the AST route. 

I could improve this in this spot, but I would need more guidance about usage of `typeOf`.

> Also worth noting that the name directive is going to behave a bit strangely for methods as it will result in the method not being namespaced under the type name. It's not likely to be a problem but might be worth a quick note?

I agree to add a note.